### PR TITLE
Update to latest PHPCompatibility.

### LINKED ISSUE #179: Whitelist: Health Check
https://wordpress.org/plugins/health-check/

I'd like to request a whitelisted of this, we've got tests running to maintain compatibility with all the versions core supports, and the lines triggered are caused by back compat checks for to determine mysql/mysqli usage.

### PR DESCRIPTION
**Pull request by octalmage**
Alright, this is a big PR and it includes a number of changes: 

* Replace PHPCompatibility with PHPCompatibilityWP, and pull in a bunch of improvements. 
* Add local development environment. 
* Add build script to copy the changes over to an SVN repo, and create a useable zip. 
* Fix JavaScript issue related to #31. 

Resolves #179. 

I also updated [Building and Testing](https://github.com/wpengine/phpcompat/wiki/Building-and-Testing) to use the Docker container.

### CODE REVIEW

**Comment by octalmage** [comment]
I'll check it out! I heard the maintainer is pretty cool.

**Comment by octalmage** [comment]
Seems to be working great! Thanks!

**Comment by jrfnl** [comment]
@octalmage If you're interested (or know of someone else who might be), we'd be happy to welcome more maintainers for the WP ruleset.

All that's involved in that is, once in a while, checking if any new back-fills/poly-fills for native PHP functionality have been added to WP and sending in PRs adding new excludes to the ruleset.

**Comment by octalmage** [comment]
Yeah of course! We’re about to start scanning a bunch of sites so I’d be happy to contribute any other rules I see.

**Comment by jrfnl** [comment]
@octalmage I look forward to your first PR ;-)

File: .travis.yml [hunk 1]
```diff
@@ -71,26 +71,9 @@ before_script:
...
-      cd ..
-      cp -r "$PLUGIN_SLUG" "/tmp/wordpress/src/wp-content/plugins/$PLUGIN_SLUG"
-      cd /tmp/wordpress/
-      cp wp-tests-config-sample.php wp-tests-config.php
-      sed -i "s/youremptytestdbnamehere/wordpress_tests/" wp-tests-config.php
-      sed -i "s/yourusernamehere/travis/" wp-tests-config.php
-      sed -i "s/yourpasswordhere//" wp-tests-config.php
-      mysql -e "CREATE DATABASE wordpress_tests;" -uroot
-      cd "/tmp/wordpress/src/wp-content/plugins/$PLUGIN_SLUG"
+      tests/install-wp-tests.sh wordpress_tests travis "" localhost $WP_VERSION false
```
Comment (octalmage) [inline .travis.yml]: Simplify this a bit by using the install-wp-tests script provided by wp-cli.

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+if [ ! -d "build" ]; then
+	svn co https://plugins.svn.wordpress.org/php-compatibility-checker build
+else
+	pushd build
+	svn up
+	popd
+fi
```
Comment (octalmage) [inline build.sh]: I felt that it was important to build the zip with the actual code found in the SVN repo.

**Pull request merged**